pkg/ansible/paramconv/paramconv.go

```diff
@@ -84,6 +91,38 @@ func ToSnake(s string) string {
 	bits := []string{}
 	n := ""
 	iReal := -1
+	count := 0
+	length := len(s)
+	for i, v := range s {
+		if i+1 < length {
+			next := s[i+1]
+			count = count + 1
+			if (v >= 'A' && v <= 'Z' && next >= 'a' && next <= 'z') || (v >= 'a' && v <= 'z' && next >= 'A' && next <= 'Z') {
+				if next >= 'A' && next <= 'Z' {
+					count = 0
+				} else if next >= 'a' && next <= 'z' && next != 's' {
+					// fmt.Printf("This is the slice: %s", s[i-count+1:i+1])
+					if _, ok := wordMapping[strings.ToLower(s[i-count+1:i+1])]; ok && i-count+1 == 0 {
+						s = s[0:i+1] + "_" + s[i+2:]
+					} else if _, ok := wordMapping[strings.ToLower(s[i-count+1:i+1])]; ok {
+						s = s[0:i-count+1] + "_" + s[i-count+1:i+1] + "_" + s[i+1:]
+					}
+					// fmt.Printf("This is the final string: %s", s)
+				} else if next >= 'a' && next <= 'z' && next == 's' {
+					if _, ok := wordMapping[strings.ToLower(s[i-count+1:i+2])]; ok && i+2 == length {
+						s = s[0:i-count+1] + "_" + s[i-count+1:i+1] + "S"
+						// fmt.Printf("\n this si the S : %s\n", s)
+					} else if _, ok := wordMapping[strings.ToLower(s[i-count+1:i+2])]; ok && i-count+1 == 0 {
+						s = s[0:i+1] + "S" + "_" + s[i+2:]
+					} else {
+						s = s[0:i-count+1] + "_" + s[i-count+1:i+1] + "S" + "_" + s[i+2:]
```

Comment:
I don't like the concatenations but I don't know of a better way. 
Using `fmt.Sprintf` seems to be about the same complexity.

`s = fmt.Sprintf("%s_%sS_%s", s[0:i-count+1], s[i-count+1:i+1], s[i+2:])`